plugins/ru.runa.gpd/src/ru/runa/gpd/extension/businessRule/BusinessRuleEditorDialog.java

```diff
@@ -0,0 +1,542 @@
+                targetCombo.setItems(new String[0]);
+                GroovyTypeSupport typeSupport = GroovyTypeSupport.get(firstVariable.getJavaClassName());
+                Operation operation = Operation.getByName(operCombo.getText(), typeSupport);
+                operCombo.setData(DATA_OPERATION_KEY, operation);
+                for (String variableName : getSecondVariableNames(firstVariable)) {
+                    targetCombo.add(variableName);
+                }
+                for (String pv : typeSupport.getPredefinedValues(operation)) {
+                    targetCombo.add(pv);
+                }
+                if (typeSupport.hasUserInputEditor()) {
+                    targetCombo.add(TypedUserInputCombo.INPUT_VALUE);
+                }
+            }
+        } catch (RuntimeException e) {
+            PluginLogger.logError(e);
+        }
+    }
+
+    @Override
+    protected void toCode() {
+        for (int i = 0; i < expressionLines.size(); i++) {
+            ExpressionLine expressionLine = expressionLines.get(i);
+            for (int j = 0; j < expressionLine.getVariableBoxes().size(); j++) {
+                boolean emptyFieldExist = expressionLine.getVariableBoxes().get(j)[0].getText().length() == 0
+                        || expressionLine.getVariableBoxes().get(j)[1].getText().length() == 0
+                                && !expressionLine.getFunctionButton().getText().equals(defaultFunction);
+                if (emptyFieldExist) {
+                    setErrorLabelText(Localization.getString("GroovyEditor.fillAll"));
+                    // we cannot construct while all data not filled
+                    return;
+                }
+            }
+        }
+        clearErrorLabelText();
+        try {
+            BusinessRuleModel model = new BusinessRuleModel();
+            if (defaultFunction != null) {
+                model.setDefaultFunction(defaultFunction);
+            }
+            for (int i = 0; i < expressionLines.size(); i++) {
+                ExpressionLine expressionLine = expressionLines.get(i);
+                IfExpr ifExpr;
+                List<Variable> firstVariables = new ArrayList();
+                List<Object> secondVariables = new ArrayList();
+                List<Operation> operations = new ArrayList();
+                List<String> logicExprsessions = new ArrayList();
+                for (int j = 0; j < expressionLine.getVariableBoxes().size(); j++) {
+                    Variable firstVariable = (Variable) expressionLine.getVariableBoxes().get(j)[0].getData(DATA_VARIABLE_KEY);
+                    String operationName = expressionLine.getOperationBoxes().get(j)
+                            .getItem(expressionLine.getOperationBoxes().get(j).getSelectionIndex());
+                    String secondVariableText = expressionLine.getVariableBoxes().get(j)[1].getText();
+                    Object secondVariable;
```

Comment:
`Object` как тип переменной не нравится, да и название `variable2` возникает из-за этого же.

Кажется, так будет лучше (вместо вязанки if'ов, связанной с инициализацией `secondVariable`):
```suggestion
                    Variable secondVariable = VariableUtils.getVariableByScriptingName(variables, secondVariableText);  
                    if (secondVariable != null && secondVariable.getJavaClassName().equals(firstVariable.getJavaClassName())) {
                        secondVariables.add(secondVariable);
                    } else {
                        secondVariables.add(secondVariableText);
                    }
```

Ниже, соответсвенно, `secondVariables.add(secondVariable);` убрать